lib/linguist/language.rb

```diff
@@ -166,36 +168,31 @@ def self.find_by_alias(name)
     #
     # Examples
     #
-    #   Language.find_by_filename('foo.rb')
-    #   # => [#<Language name="Ruby">]
+    #   Language.find_by_filename('Cakefile')
+    #   # => [#<Language name="CoffeeScript">]
     #
     # Returns all matching Languages or [] if none were found.
     def self.find_by_filename(filename)
       basename = File.basename(filename)
-
-      # find the first extension with language definitions
-      extname = FileBlob.new(filename).extensions.detect do |e|
-        !@extension_index[e].empty?
-      end
-
-      (@filename_index[basename] + @extension_index[extname]).compact.uniq
+      @filename_index[basename]
```

Comment:
Is there anything I need to do in this pull request for that? Add a warning somewhere or a note in comments maybe?